### DISCUSSION TITLE
Cucumber gherkin keywords are not displaying in Allure report

### DISCUSSION THREAD

**Discussion by Sathish-sk1** [mmisty/cypress-allure-adapter#91] [Q&A] [answered]
In older shelex allure report the cucumber gherkin keywords are displaying properly and had very good readability

![image](https://github.com/mmisty/cypress-allure-adapter/assets/150774370/5684fb19-195a-4f54-8f23-cd0f2cb539f1)

But in @mmisty/cypress-allure-adapter the gherkin keywords are not displaying properly and making very hard to read

![image](https://github.com/mmisty/cypress-allure-adapter/assets/150774370/de77343a-5488-4812-9470-482528ec88e9)

![image](https://github.com/mmisty/cypress-allure-adapter/assets/150774370/4c5ba958-76c0-4835-b292-b80948644f30)

I have provided screenshots of same scenario in both she;ex-allure and @mmisty/cypress-allure-adapter.

package.json
 "@mmisty/cypress-allure-adapter": "^0.16.1"
 
 config.js
  env: {
     allure: true,
      allureResults: 'cypress/reports/allure-results',
      allureCleanResults: true,
      allureShowDuplicateWarn: true,
      allureShowTagsInTitle: true,
      allureLogCyCommands: true,
    }
    
    @mmisty , Please help me here

**Comment by Sathish-sk1** [comment]
Hi @mmisty , any update on this?

**Reply by mmisty** [reply]
@Sathish-sk1 hi, 
I didn't have chance to look at it yet. 
Could you please provide the implementation of gherkin steps that you have on screenshot

**Reply by Sathishtes** [reply]
Hi @mmisty , Please find the gherkin steps here
Scenario: Verify if user 1 cannot modify or delete games booked by user 1
    When the user login as 'data' with 'data' details
    Then the url contains '/data/data'
    And the field 'data' contains 'data'
    When the user clicks on clubs and events 'data' as a parent
    Then the field 'data' contains 'data'
    When the user clicks on 'Amend Bookings' in the field 'data'
    Then the field 'data' contains 'data'
    And the user verifies the count in the field 'data' is equal to 4
    And the user encounters an error message when selecting index 'data' for the clubB 'data' with the specified page heading 'data'
    And the user encounters an error message when selecting index 'data' for the clubB 'data' with the specified page heading 'data'
    And the field 'data' contains 'data'

the Given, When  and Then keywords are partially broken in the report

**Reply by mmisty** [reply]
Hi @Sathishtes , thanks!
could you please also send the implementation of these steps i.e. cypress commands that are being executed within the steps.

**Reply by Sathish-sk1** [reply]
Hi @mmisty, please find the implementation here
Given('the user login as {string} with {string} details', (user1, user2) => {
  cy.get(selectors.acd_login_user_1_title)
    .should('be.visible')
    .select(data.user_1_details[`${user1}_title`]);
  cy.get(selectors.acd_login_user_1_firstname)
    .should('be.visible')
    .type(data.user_1_details[`${user1}_firstname`]);
  cy.get(selectors.acd_login_user_1_surname)
    .should('be.visible')
    .type(data.user_1_details[`${user1}_surname`]);
  cy.get(selectors.acd_login_user_2_firstname)
    .should('be.visible')
    .type(data.user_2_details[`${user2}_firstname`]);
  cy.get(selectors.acd_login_user_2_surname)
    .should('be.visible')
    .type(data.user_2_details[`${user2}_surname`]);
  cy.get(selectors.acd_login_birth_day)
    .should('be.visible')
    .select(data.user_2_details[`${user2}_birth_day`]);
  cy.get(selectors.acd_login_birth_month)
    .should('be.visible')
    .select(data.user_2_details[`${user2}_birth_month`]);
  cy.get(selectors.acd_login_birth_year)
    .should('be.visible')
    .select(data.user_2_details[`${user2}_birth_year`]);
  cy.get(selectors.acd_login_user_1_btn).should('be.visible').click();
});

Given('the page contains these values', (dataTable) => {
  dataTable.hashes().forEach((elem) => {
    cy.get(selectors[elem.Element])
      .should('be.visible')
      .and('contain', elem.Value);
  });
});

Given(
  'the field {string} contains {string}',
  (elementSelector, assertionValue) => {
    const array = assertionValue.split(',');
    if (assertionValue) {
      for (let item in array) {
        cy.contains(array[item]);
        cy.get('body').then((body) => {
          if (body.find(selectors[elementSelector]).length > 0) {
            cy.get(selectors[elementSelector]).should((elementSelector) => {
              expect(elementSelector).to.contain(array[item]);
            });
          } else {
            cy.get(selectors[elementSelector])
              .scrollIntoView()
              .should((elementSelector) => {
                expect(elementSelector).to.contain(array[item]);
              });
          }
        });
      }
    }
  }
);

Given('the user clicks on concert and games {string} as a admin', (Value) => {
  cy.get(selectors.acd_concert_events_layout)
    .find(selectors.acd_concert_events_admin_card)
    .each(($e1) => {
      const textClub = $e1.find(selectors.acd_concert_events_header).text();
      if (textClub.includes(Value)) {
        cy.wrap($e1)
          .find(selectors.acd_concert_events_open_for_booking_btn)
          .click({ force: true });
      }
    });
});

Given('the user clicks on {string}', (elementSelector) => {
  cy.get('body').then((body) => {
    if (body.find(selectors[elementSelector]).length > 0) {
      cy.get(selectors[elementSelector]).click();
    } else {
      cy.contains(elementSelector).click();
    }
  });
});

Given(
  "the user verifies the count in the field {string} is equal to {int}",
  (elementSelector, count) =>
    cy.get(selectors[elementSelector]).should("have.length", count)
);

Given(
  "the user encounters an error message when selecting index {string} for the concertB {string} with the specified page heading {string}",
  (index, eventType, assertionValue) => {
    const card = selectors.acd_concert_card;
    cy.contains(eventType).then((ele) => {
      cy.get(ele)
        .eq(0)
        .parents(card)
        .find(selectors.acd_user_1_booking_cell)
        .eq(index)
        .click();
      cy.get(selectors.acd_concert_events_edit_or_delete_page).should(
        "contain",
        assertionValue
      );
      cy.contains(
        "The booking cannot be deleted."
      ).should("be.visible");
      cy.get(selectors.acd_concert_event_x_btn).click();
    });
  }
);

**Reply by Sathish-sk1** [reply]
Hello @mmisty , in Shelex-allure I can avoid logging both username and password in Jenkins report by using allureAvoidLoggingCommands: ['login'], but in mmisty report I've tried this allureSkipSteps: "login" but still it displays both username and password. Not avoiding the logs of the particular command

**Reply by Sathish-sk1** [reply]
Hi @mmisty , any fix for this issue?

**Reply by Sathish-sk1** [reply]
Hi @mmisty , @DamirPecnik the issue is still persisting, could you please update on this issue?

**Reply by mmisty** [reply]
Hi @Sathish-sk1 
I cannot reproduce the issue, if you could create reproducible example repo I would be able to fix that.
You can start creating repo with these examples https://github.com/mmisty/cypress-allure-adapter-example/tree/main/examples

**Reply by Sathish-sk1** [reply]
Hi @mmisty ,  in mmisty allure report I've tried using allureSkipSteps: "login" to hide login credentials but it is not skipping this 'login' in report but it displays both username and password in the report. It is not avoiding the logs of the particular command. Please suggest is any workaround for avoiding logs for cypress default and custom commands.

**Reply by mmisty** [reply]
@Sathish-sk1 hello, if the issue is still actual could you please try using 'allureSkipCommands' env variable

**Reply by mmisty** [reply, marked as answer]
Please try version 2.1.0, may be fixed there.

**Reply by Sathishtes** [reply]
Hi @mmisty , the issue is fixed in 2.1.0, thanks
